Jenkinsfile

```
@@ -27,18 +28,8 @@ def generateParallelSteps(labels) {
         parallelNodes[label] = {
             node(label) {
                 if (isUnix()) {
-                    sh '''
-                        uname -a
-                        if test -e /etc/os-release; then
-                            cat /etc/os-release
-                        fi
-                        if test -e /proc/cpuinfo; then
-                            cat /proc/cpuinfo
-                        fi
-                        if test -e /proc/meminfo; then
-                            cat /proc/meminfo
-                        fi
-                    '''
+                    checkout scm
+                    sh 'bash ./checks.sh '+label
```

Why not make the script executable and call it without the invocation of `bash`?  Is there a concern that we may run FreeBSD or OpenBSD or macOS agents where `/bin/bash` does not necessarily exist?

This suggested change is optional.  You're welcome to reject it if you would rather not make the file executable.
```suggestion
                    sh './checks.sh '+label
```

---

if I remember correctly, the main concern is for the file rights (execution) and then the fact of potential nonexistence of bash as you mentioned.

---

What @smerle33 is correct: `bash` is a requirement for the script (as per its shebang at least) so it's now explicit in the pipeline (Particularly since the `sh` pipeline instruction is using `sh` by default) call and the execution bit is not always guaranteed to be applied on the shell script (depends on how it is retrieved and on the underlying filesystem).